src/main/scala/org/clulab/wm/eidos/groundings/grounders/SRLCompositionalGrounder.scala

```diff
@@ -187,87 +185,105 @@ class SRLCompositionalGrounder(name: String, domainOntology: DomainOntology, w2v
-                }
-            }
-        Seq(PredicateGrounding(predicateTuple))
-
-      case predicates =>
-        // Yes predicates
-        // First try to exact match entire mention before splitting up into args and preds
-        // Keep longest exact match, then try to ground remaining content
-
-        // Try to exact match entire token interval
-        val mentionWords: Array[String] =
-          for (w <- s.words.slice(tokenInterval.start, tokenInterval.end))
-          yield w.toLowerCase
-        val embeddings = SRLCompositionalGrounder.processOrConceptBranches.flatMap(conceptEmbeddingsSeq(_))
-        val exactMatches = exactMatchForPreds(mentionWords, embeddings).getOrElse(Seq.empty) //the max overlap match
-        val exactMatch = newOntologyGrounding(exactMatches)
-        // remove exact match overlap from remaining content
-        val remainingContent = if (exactMatches.nonEmpty) mentionWords.toSet.diff(StringUtils.afterLast(exactMatches.head.name.toLowerCase.replaceAll("_", " "), '/', true).split(" ").toSet).toArray else mentionWords
-
-        val exactMatchPredicateTuple = if (exactMatches.isEmpty) PredicateTuple(emptyOntologyGrounding, emptyOntologyGrounding, emptyOntologyGrounding, emptyOntologyGrounding, tokenInterval.toSet) else exactMatches.head.branchOpt match {
-          case Some(SRLCompositionalGrounder.CONCEPT) => PredicateTuple(exactMatch, emptyOntologyGrounding, emptyOntologyGrounding, emptyOntologyGrounding, tokenInterval.toSet)
-          case Some(SRLCompositionalGrounder.PROCESS) => PredicateTuple(emptyOntologyGrounding, emptyOntologyGrounding, exactMatch, emptyOntologyGrounding, tokenInterval.toSet)
+    def emptyOrBranchToPredicateTuple(ontologyGrounding: OntologyGrounding): PredicateTuple = {
+      if (ontologyGrounding.isEmpty)
+        // If there's nothing here, return an empty grounding.
+        PredicateTuple(emptyOntologyGrounding, emptyOntologyGrounding, emptyOntologyGrounding, emptyOntologyGrounding, tokenInterval.toSet)
+      else {
+        val branch = ontologyGrounding.grounding.head.branchOpt.get
+
+        branch match {
+          case SRLCompositionalGrounder.CONCEPT =>
+            PredicateTuple(ontologyGrounding, emptyOntologyGrounding, emptyOntologyGrounding, emptyOntologyGrounding, tokenInterval.toSet)
+          case SRLCompositionalGrounder.PROCESS =>
+            PredicateTuple(emptyOntologyGrounding, emptyOntologyGrounding, ontologyGrounding, emptyOntologyGrounding, tokenInterval.toSet)
         }
-        // If there is no content left, stop here.
-        // Just return the exact match
-        if (remainingContent.isEmpty) {
-          Seq(PredicateGrounding(exactMatchPredicateTuple))
+      }
+    }
+
+    def groundWithoutPredicates(sentenceHelper: SentenceHelper): Seq[PredicateGrounding] = {
```

Comment:
I really like separating the without/with predicates conditions like this!

Reply:
Thanks for being flexible.  It was mostly to cut down the length.  The method/paragraph was too long for me.